wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraInitialConnectionStrategyPlugin.java

```diff
@@ -110,12 +143,8 @@ public Connection connect(
 
     final RdsUrlType type = this.rdsUtils.identifyRdsType(hostSpec.getHost());
 
-    if (!type.isRdsCluster()) {
-      // It's not a cluster endpoint. Continue with a normal workflow.
-      return connectFunc.call();
-    }
-
-    if (type == RdsUrlType.RDS_WRITER_CLUSTER) {
+    if (type == RdsUrlType.RDS_WRITER_CLUSTER
+        || isInitialConnection && this.verifyOpenedConnectionType == VerifyOpenedConnectionType.WRITER) {
```

Comment:
I think we need an extra set of brackets, otherwise we will still verify when using a writer cluster and it is not an initial connection
```suggestion
        || (isInitialConnection && this.verifyOpenedConnectionType == VerifyOpenedConnectionType.WRITER)) {
```